frontend/src/bundles/common/components/button/button.tsx

```diff
@@ -1,10 +1,95 @@
-type Properties = {
-    label: string;
-    type?: 'button' | 'submit';
+import clsx from 'clsx';
+import { type ReactNode } from 'react';
+
+import {
+    type IconName,
+    ButtonSize,
+    ButtonType,
+    ButtonVariant,
+    IconSize
+} from '~/bundles/common/enums/enums';
+import { type ValueOf } from '~/bundles/common/types/types';
+
+import { Icon } from '../icon/icon';
+import styles from './styles.module.scss';
+
+const sizes: Record<ButtonSize, string> = {
+    small: styles.size__small,
+    medium: styles.size__medium,
+};
+
+const variants: Record<ButtonVariant, string> =  {
+    default: styles.button__base,
+    primary: styles.button__primary,
+    outlined: styles.button__outlined,
+    ghost: styles.button__ghost
+};
+
+type BaseButtonProperties = {
+    children?: ReactNode;
+    type?: ValueOf<typeof ButtonType>;
+    variant?: ValueOf<typeof ButtonVariant>;
+    size?: ValueOf<typeof ButtonSize>;
+    onClick?: React.MouseEventHandler<HTMLButtonElement>;
+    isDisabled?: boolean;
+    isFluid?: boolean;
+    className?: string;
+    appendedIcon?: ValueOf<typeof IconName>;
```

Comment:
Do not restrict component to use only icons from the predefined list. Just pass `AppendedComponent?: ReactNode` prop